src/main/java/com/opentable/kafka/util/OffsetMetrics.java

```diff
@@ -77,10 +95,12 @@
 
     private final String metricPrefix;
     private final MetricRegistry metricRegistry;
-    private final String groupId;
     private final Collection<String> topics;
     private final Duration pollPeriod;
     private final OffsetMonitor monitor;
+    private final boolean expectKafkaManagedOffsets;
+    /** Topic -> partition -> offset. */
+    private final Function<String, Map<Integer, Long>> offsetsSupplier;
```

Comment:
FWIW, I chose to assume dense partitions and used `long[]` in other places.

Reply:
Cool, good context.